Warships don't detect own ports as portspaces

Noordfrees opened the pull request:
**Type of change**
Bugfix

**Issue(s) closed**
Fixes the graphics glitch from #6085

**New behavior**
Warships already don't detect own ports as port spaces, but now they also *stop* detecting them when the port-under-construction is completed.

**Possible regressions**
Warship port space detection and forgetting

Noordfrees committed:
Warships don't detect own ports as portspaces

Noordfrees committed:
'src/logic/map_objects/tribes/ship.cc' was automatically formatted.

Noordfrees committed:
Merge branch 'protected/navalwarfare' into warship-search-ignore-own-port

tothxa reviewed (commented):
Code looks OK with one possible improvement.

However in testing, ships still/again stop 3 times at the same port space. So a check and update of `expedition->seen_port_buildspaces` is missing somewhere. It should however still allow updating the portspace info, and only prevent stopping and message sending.

Noordfrees committed:
Merge branch 'protected/navalwarfare' of github.com:widelands/widelands into warship-search-ignore-own-port

Noordfrees committed:
address revie

Noordfrees committed:
'src/logic/map_objects/tribes/ship.cc' was automatically formatted.

Noordfrees committed:
Merge branch 'protected/navalwarfare' into warship-search-ignore-own-port

Noordfrees committed:
Merge branch 'protected/navalwarfare' of github.com:widelands/widelands into warship-search-ignore-own-port

Noordfrees committed:
More refactoring as requested

Noordfrees committed:
3 files were automatically formatted.

tothxa committed:
Unify portspace search code for expeditions and warships
Make the last spotted portspace the target
Enable build port button in expeditions also when moving

tothxa committed:
clang-tidy

tothxa committed:
send message on stopping ship

tothxa committed:
revert incompatible changes

tothxa committed:
simplify

tothxa committed:
clang-tidy

Noordfrees committed:
Merge branch 'protected/navalwarfare' into warship-search-ignore-own-port

tothxa committed:
- use member function instead of std::function var
 - update owner of seen but unusable portspaces too
 - add options for stopping/reporting (not used yet)

tothxa committed:
- do not expose expedition_->seen_port_buildspaces, replace with specific checks
 - do not cache seen portspaces in InteractivePlayer (implies saveloading format change)
 - allow invasion on any seen port space, not just the primary

tothxa committed:
clang-tidy with actual bugfix and restore mistakenly deleted case

tothxa committed:
fix mixed up variable

tothxa committed:
warships show all seen port spaces equally, according to battle state

tothxa committed:
optimisation

Noordfrees committed:
Merge branch 'protected/navalwarfare' into warship-search-ignore-own-port

Noordfrees committed:
2 files were automatically formatted.

tothxa reviewed (approved):
I guess we did review each other's changes, so that can be enough?
